### ISSUE TITLE
is symbol type permitted as a computed property name in an interface ?

### ISSUE THREAD

**Issue by sciencefunq** [microsoft/TypeScript#62064]
### 🔎 Search Terms

computed property name , interface 

### 🕗 Version & Regression Information

- This bug appears in typescript@5.8

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.7.3#code/DYUwLgBAzgngtgLlnARge2BAvBAyvdYACgEoBuAKAEsA7MEAJwDMBDAYxAgDE00IBvCBQgQA2sgC6CaGAa0A5mQgB6ZRDAALKlAjBanBi20gdLGhEYM0DCLXVQAAgFYAdAHYIAGhQBXSABM0EwgaNEhDYwgzCwYrGzswR1cADhcAZgoAXyA

### 💻 Code

```ts
let sym:symbol = Symbol();
interface Foo { 
  [sym]: string; // this line raises an error in ts@5.7 ,but does not raise an error in ts@5.8.3
}
```

### 🙁 Actual behavior

After I upgraded my typescipt version to 5.8.3, this error disappears. 

### 🙂 Expected behavior

In typescript versions before 5.8, the code in  interface Foo raised an error:  _error TS1169: A computed property name in an interface must refer to an expression whose type is a literal type or a 'unique symbol' type.

### Additional information about the issue

 is it a error or a new feature ?

**Comment by MartinJohns** [comment]
If you already noticed the behavior changed between version 5.7 and 5.8, why not check the release notes for relevant entries?

**Comment by RyanCavanaugh** [comment]
https://devblogs.microsoft.com/typescript/announcing-typescript-5-8/#preserved-computed-property-names-in-declaration-files
